tests/EventTests/EventBuilderTest.php

```diff
@@ -216,118 +418,157 @@ public function testCreateConversionEventWithAttributesNoValue()
-                        'variationId' => '7722370027',
-                        'isLayerHoldback' => false
-                    ]
-                ]]
+                ]
             ],
             'POST',
             ['Content-Type' => 'application/json']
         );
+        
+        $userAttributes = [
+            'invalid_attribute'=> 'sorry_not_sorry'
+        ];
         $logEvent = $this->eventBuilder->createConversionEvent(
             $this->config,
             'purchase',
             ['7716830082' => '7722370027'],
             $this->testUserId,
-            null,
-            array('revenue' => 42)
+            $userAttributes,
+            null
         );
 
+        $logEvent = $this->fakeParamsToReconcile($logEvent);
         $this->assertEquals($expectedLogEvent, $logEvent);
     }
 
-    public function testCreateConversionEventWithAttributesWithValue()
+    public function testCreateConversionEventNoAttributesWithValue()
     {
         $expectedLogEvent = new LogEvent(
-            'https://logx.optimizely.com/log/event',
+            'https://logx.optimizely.com/v1/events',
             [
-                'projectId' => '7720880029',
-                'accountId' => '1592310167',
-                'visitorId' => 'testUserId',
+                'project_id' => '7720880029',
+                'account_id' => '1592310167',
                 'revision' => '15',
-                'clientEngine' => 'php-sdk',
-                'clientVersion' => '1.2.0',
-                'isGlobalHoldback' => false,
-                'timestamp' => time() * 1000,
-                'eventFeatures' => [
-                    [
-                        'name' => 'revenue',
-                        'type' => 'custom',
-                        'value' => 42,
-                        'shouldIndex' => false
-                    ],
-                    [
-                        'name' => 'non-revenue',
-                        'type' => 'custom',
-                        'value' => 'definitely',
-                        'shouldIndex' => false
+                'client_name' => 'php-sdk',
+                'client_version' => '1.2.0',
+                'visitors'=> [[
+                  'attributes'=> [],
+                  'visitor_id'=> 'testUserId',
+                  'snapshots'=> [[
+                    'decisions'=> [[
+                      'variation_id'=> '7722370027',
+                      'experiment_id'=> '7716830082',
+                      'campaign_id'=> '7719770039'
+                    ]],
+                    'events'=> [[
+                      'timestamp'=> $this->timestamp,
+                      'entity_id'=> '7718020063',
+                      'uuid'=> $this->uuid,
+                      'key'=> 'purchase',
+                      'revenue' => 42,
+                      'value'=> 13.37,
+                      'tags' => [
+                        'revenue' => 42,
+                        'value' => '13.37'
```

Comment:
@oakbani  please remove '' as you are passing float value.

Reply:
If value is passed as numeric string, the
value tag will return float value but the tags[] will return values in original format as per document. 

"If the value is stringified, we need to make sure that the value sent is a double/float, but the event tag value should still have it as a string. Also make sure that `revenue` still works as expected."